base_layer/core/src/chain_storage/lmdb_db/lmdb_db.rs

```diff
@@ -2495,6 +2533,17 @@ impl BlockchainBackend for LMDBDatabase {
                 value: public_key.to_hex(),
             })
     }
+
+    fn fetch_template_registrations(&self, from_height: u64) -> Result<Vec<TemplateRegistration>, ChainStorageError> {
+        let txn = self.read_transaction()?;
+        lmdb_filter_map_values(&txn, &self.template_registrations, |tr: TemplateRegistration| {
```

Comment:
We can optimise this query by making using a compound key <reg_height, hash> - but we can optimise this later